ParkingZone/Areas/Admin/Controllers/ParkingZonesController.cs

```diff
@@ -136,19 +131,19 @@ public async Task<IActionResult> Delete(long? id)
         [ValidateAntiForgeryToken]
         public async Task<IActionResult> DeleteConfirmed(long id)
         {
-            var parkingZone = await _context.ParkingZones.FindAsync(id);
+            var parkingZone = _repository.GetById(id);
             if (parkingZone != null)
             {
-                _context.ParkingZones.Remove(parkingZone);
+                _repository.Delete(parkingZone);
             }
 
-            await _context.SaveChangesAsync();
             return RedirectToAction(nameof(Index));
```

Comment:
```suggestion
            return NotFound();
```

Reply:
because parking Zone is null here

Reply:
return RedirectToAction(nameof(Index)); ,  that action provides redirection of index , if we return notfound() the index does not get back

Reply:
yes you are right, but it would be logical when redirecting after deleting. It looks weird redirecting without deleting. Here i meant, if parkingZone == null return NotFound();